service/topology_state_machine.hh

```diff
@@ -92,8 +87,14 @@ struct topology {
     // operation untill the node becomes normal
     std::unordered_map<raft::server_id, request_param> req_param;
 
+    // The ID of the last introduced CDC generation.
     std::optional<cdc::generation_id_v2> current_cdc_generation_id;
 
+    // This is the UUID used to access the data of a new CDC generation introduced
+    // e.g. when a new node bootstraps, needed in `commit_cdc_generation` transition state.
+    // It's used as partition key in CDC_GENERATIONS_V3 table.
+    std::optional<utils::UUID> new_cdc_generation_data_uuid;
```

Comment:
Not related to this patch, but I think for 2-phase generation switching, we should switch cdc generation id to timeuuid and avoid a separate timestmap to describe the generation.

Reply:
A similar observation applies to stream ids. Instead of using "random" bits we could round up generation timestamp to 48 bits (i.e. have second-precision generation timestamp, we always generate it in the future anyway) and store it directly in the stream id.

Reply:
> Not related to this patch, but I think for 2-phase generation switching, we should switch cdc generation id to timeuuid and avoid a separate timestamp to describe the generation.

The separation of UUID / timestamp allows inserting CDC generation data - which takes a long time - before choosing the generation's timestamp which we want to propagate quickly.